docs/user-docs/build_grub.md

```diff
@@ -0,0 +1,199 @@
+Building GRUB from sources
+==========================
+
+At the time of writing (August 2024) EFI and legacy changes are on different
+branches in the same repository and neither builds on top of the other in
+terms of commit history.  Either clone the repository in full and switch
+between branches to build both versions or use shallow clones as shown below.
+
+If your system isn't setup for development, you might want to use
+[trenchboot-sdk] Docker container.  It can be run like this to perform a build:
+
+```
+docker run --rm -it -v "$PWD:$PWD" -w "$PWD" \
+           -e HOME="$PWD/home" \
+           --user "$(id -u):$(id -g)" \
+           ghcr.io/trenchboot/trenchboot-sdk:master /bin/bash
+```
+
+Setting `$HOME` is necessary because GRUB will try to use `ccache` which fails
+if there is no `$HOME`.  To install GRUB for legacy boot, the container might to
+be started as `root` (inside of the container, don't prepend `sudo` to the
+command):
+
+```
+docker run --rm -it -v "$PWD:$PWD" -w "$PWD" \
+           --user root:root \
+           ghcr.io/trenchboot/trenchboot-sdk:master /bin/bash
+```
+
+[trenchboot-sdk]: https://github.com/TrenchBoot/trenchboot-sdk
+
+## EFI version
+
+### Build
+
+```bash
+# check out the sources
+git clone --branch grub-sl-2.12-v9 --depth 1 https://github.com/TrenchBoot/grub.git
+cd grub
+
+# make the checkout buildable
+./bootstrap
+
+# configure
+mkdir build
+cd build
+../configure --with-platform=efi --target=x86_64
+
+# build
+make -j$(nproc)
+```
+
+### Setup
+
+```bash
+# also from inside build/ directory
+echo 'configfile ${cmdpath}/grub.cfg' > embedded.cfg
+./grub-mkimage -O x86_64-efi -o grubx64.efi -p / -c embedded.cfg -d grub-core all_video boot btrfs cat chain configfile echo efifwsetup efinet ext2 fat font gfxmenu gfxterm gzio halt hfsplus iso9660 jpeg loadenv loopback lvm mdraid09 mdraid1x minicmd normal part_apple part_msdos part_gpt password_pbkdf2 png reboot regexp search search_fs_uuid search_fs_file search_label serial sleep syslinuxcfg test tftp video xfs backtrace http linux usb usbserial_common usbserial_pl2303 usbserial_ftdi usbserial_usbdebug keylayouts at_keyboard multiboot2
```

Comment:
Please wrap this line.

Reply:
https://github.com/TrenchBoot/TrenchBoot.github.io/compare/97941c079ed7b202e52b9bc265febac7b8afd04f..1a18d56ac6cad3dcb05e6dddf5ba68cdc918abfd#diff-2224c42e305121241cb783844ec6ce88b2a5b9279ee8acd098a0b63a03a61229R59-R67